frontend/services/spotify-api-service.js

```diff
@@ -23,96 +23,61 @@ class SpotifyApiService {
     })
   }
 
-  isValidSpotifyUri(uri) {
```

Comment:
Thanks for the work on this! After another pass, it looks like the validation rework expands the scope beyond the original feature and contributes to the PR’s size, which makes thorough review difficult. To keep this PR focused and reviewable, could we revert the validation changes here and address that bug in a separate, dedicated PR? That will let us define requirements, write targeted tests, and keep each PR small and easy to review. Happy to help outline the split and identify logical breakpoints.